### ISSUE TITLE
New layout, song/album/progress bar doesn't change. 

### ISSUE THREAD

**Issue by lovelydumpling** [pvrs12/Anesidora#80]
Whenever I hit skip to play a new song or change stations, the new song plays but the title/album art doesn't change until I close and re-open the player. The song's progress bar empties for the new song but also doesn't progress unless I re-open the player. 

Just checked, doesn't change even with the song changing organically. Actually, noticing the progress bar doesn't move at all while it's open.  

This is on Brave/Chromium. I'll test it with Firefox real quick. 

EDIT: Doesn't appear to happen on Firefox, v2.1.2. 

EDIT 2: Updated Firefox to v2.1.3. The bug happens there. So it appears to be a change in the newest version.

EDIT 3: Old layout does this too. Also, the play button disappears after you click it in old layout. Not sure if old layout is still being "supported" though.

**Comment by hucario** [comment]
I'll look into this.

**Comment by pvrs12** [comment]
I think this can be fixed easily by calling drawPlayer in the timeUpdate hook

________________________________

**Comment by hucario** [comment]
I thought it already was. If it isn't I'll implement it

**Comment by hucario** [comment]
alright I found the issue:
I'm a little bit dumb

what's wrong with this code?
hint: it's in background.js already
```js
background.callbacks.updatePlayer.forEach((e) => {
	try {
		e();
	} catch(b) {
		background.callbacks.updatePlayer.splice(background.callbacks.updatePlayer.indexOf(e), 1);
	}
});
```
that's right
it's already in the background
so background
_doesn't exist_
I just need to yeet all the `background.`:
```js
callbacks.updatePlayer.forEach((e) => {
	try {
		e();
	} catch(b) {
		callbacks.updatePlayer.splice(callbacks.updatePlayer.indexOf(e), 1);
	}
});
```

anyways yeah so I'm submitting a patch

**Comment by pvrs12** [comment]
Late to the party... This should now be merged and in the newest version (v2.1.4)

**Comment by lovelydumpling** [comment]
This seems to work... about half the time. Half the time when I change stations, skip songs, or play to the end of a song, the song title/album art changes. The other half, it doesn't, unless I skip/change stations/play to the end again or close and reopen the player. It weirdly enough seems to be consistently working every _other_ time. 

The playhead does work though at all times now. 

EDIT: Weird, as of this morning its working pretty consistently now. 

EDIT 2: Tested a little more, it only failed to change once when trying several things with it. It seems the issue is mostly resolved. It can just miss it sometimes.

**Comment by pvrs12** [comment]
@lovelydumpling I think this was due to one last missing `background.callbacks` here https://github.com/pvrs12/Anesidora/blob/51ef136e212f5a0957b25b4c6fa23b1231dd413f/common/js/background.js#L153

**Comment by hucario** [comment]
Damn, I thought I did a search/replace. I'll fix it.

**Comment by pvrs12** [comment]
Oh crap. I forgot the second half of that comment. I fixed it in 2.2.0

________________________________

**Comment by lovelydumpling** [comment]
Working flawlessly in 2.2.0! Also the media API thing is SUPER useful, media keys have been working in Firefox but not Chrome/ium for me for a while. It's so nice to have this functionality again. 

I suppose it's safe to close the issue now.
